### ISSUE TITLE
 File thin snapshots not getting deleted even with `deletionPolicy: Delete` 

### ISSUE THREAD

**Issue by Ulrar** [LINBIT/linstor-server#379]
Hi,

This is a repost of https://github.com/piraeusdatastore/piraeus-operator/issues/564 as I was advised there that the issue seems to be linstor not deleting the file, and the issue should be opened here.

When using file thin (on Talos) to create snapshots, it seems the underlying files are not getting deleted with the VolumeSnapshot object : 

```yaml
---
apiVersion: snapshot.storage.k8s.io/v1
kind: VolumeSnapshotClass
metadata:
  name: piraeus-snapshots
  annotations:
    snapshot.storage.kubernetes.io/is-default-class: "true"
driver: linstor.csi.linbit.com
deletionPolicy: Delete
```

```yaml
---
apiVersion: snapshot.storage.k8s.io/v1
kind: VolumeSnapshot
metadata:
  name: data-volume-snapshot-test-1
  namespace: zigbee2mqtt
spec:
  volumeSnapshotClassName: piraeus-snapshots
  source:
    persistentVolumeClaimName: zigbee2mqtt-data
```

```
Name:         data-volume-snapshot-test-1
Namespace:    zigbee2mqtt
Labels:       <none>
Annotations:  <none>
API Version:  snapshot.storage.k8s.io/v1
Kind:         VolumeSnapshot
Metadata:
  Creation Timestamp:  2023-11-17T10:47:48Z
  Finalizers:
    snapshot.storage.kubernetes.io/volumesnapshot-as-source-protection
    snapshot.storage.kubernetes.io/volumesnapshot-bound-protection
  Generation:        1
  Resource Version:  11491114
  UID:               2ca5756d-3ecd-4b7f-a3cc-11b6b08c8259
Spec:
  Source:
    Persistent Volume Claim Name:  zigbee2mqtt-data
  Volume Snapshot Class Name:      piraeus-snapshots
Status:
  Bound Volume Snapshot Content Name:  snapcontent-2ca5756d-3ecd-4b7f-a3cc-11b6b08c8259
  Creation Time:                       2023-11-17T10:47:50Z
  Ready To Use:                        true
  Restore Size:                        100Mi
Events:
  Type    Reason            Age   From                 Message
  ----    ------            ----  ----                 -------
  Normal  CreatingSnapshot  63s   snapshot-controller  Waiting for a snapshot zigbee2mqtt/data-volume-snapshot-test-1 to be created by the CSI driver.
  Normal  SnapshotCreated   60s   snapshot-controller  Snapshot zigbee2mqtt/data-volume-snapshot-test-1 was successfully created by the CSI driver.
  Normal  SnapshotReady     60s   snapshot-controller  Snapshot zigbee2mqtt/data-volume-snapshot-test-1 is ready to use.
```

File is there on the node : 
```
pvc-e57930e5-6772-41e4-8c98-99105b77970a_00000_snapshot-2ca5756d-3ecd-4b7f-a3cc-11b6b08c8259.img
```

Then I delete the snapshot : 
```
volumesnapshot.snapshot.storage.k8s.io "data-volume-snapshot-test-1" deleted
```

And the file is still there : 
```
pvc-e57930e5-6772-41e4-8c98-99105b77970a_00000_snapshot-2ca5756d-3ecd-4b7f-a3cc-11b6b08c8259.img
```

I tried waiting and it doesn't seem to go anywhere, I have to create a job with a hostPath to go and delete the file on the node (since Talos has no way to delete a file through it's API).

File itself looks good, I tried downloading it and it can be mounted (if forcing the filesystem to ignore drbd) as expected.
I thought having `DeletionPolicy: Delete` in the class meant the underlying file would get deleted but it seems not, is there something else needed to get rid of the snapshots once done with them ?
Thanks

**Comment by WanzenBug** [comment]
To give a bit of linstor-specific input:

A storage pool of type `FILE` seems to have a broken snapshot deletion implementation.

Namely, if you run

```
$ linstor snapshot delete res1 snap1
```

You get a success message, but on the file system the file still exists.

**Comment by ghernadi** [comment]
Bug found and fixed. We are preparing a bugfix release right now, which should also include a fix for this.

**Comment by Ulrar** [comment]
That's great news, thanks to you both for the quick help and fix!

**Comment by WanzenBug** [comment]
This was fixed with 15a3a09d70777959b120101bbdaf1faf79f01faa back in 1.25.1.  @ghernadi let's close this one.
